File: 0508-Most Frequent Subtree Sum/0508-Most Frequent Subtree Sum.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    vector<int> findFrequentTreeSum(TreeNode* root) {
        treeSum(root);

        vector<int> result;
        for (auto t : table)
        {
            if (t.second == maxCount)
            {
                result.push_back(t.first);
            }
        }
        
        return result;
    }
    
private:
    unordered_map<int, int> table;
    int maxCount;
    int treeSum(TreeNode* root)
    {
        if (root == nullptr)
        {
            return 0;
        }
        
        int left = treeSum(root->left);
        int right = treeSum(root->right);
        int sum = left + root->val + right;
        if (++table[sum] > maxCount)
        {
            maxCount = table[sum];
        }
        return sum;
    }
};

File: 0501-0600/0508-Most Frequent Subtree Sum/0508-Most Frequent Subtree Sum.py
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

import collections
class Solution:
    def findFrequentTreeSum(self, root: TreeNode) -> List[int]:
        table = collections.Counter()
        maxCount = 0
        def treeSum(root):
            nonlocal maxCount
            if root is None:
                return 0
            currSum = treeSum(root.left) + root.val + treeSum(root.right)
            table[currSum] += 1
            if table[currSum] > maxCount:
                maxCount = table[currSum]
            return currSum
        treeSum(root)
        return [k for k, v in table.items() if v == maxCount]
            